docs/authentication/hosted-add-authentication.mdx

```diff
@@ -0,0 +1,267 @@
+
+![Hosted Web Configuration](../images/hosted-web-authentication.png) 
+
+## Configure your own client / How it works
+
+### Authorization
+
+If you prefer to build your own client, you will need to start with the **Authorization Endpoint** found in your application's **External Protocol** tab. 
+
+This url should look like the following:
+
+```bash title="Authorization Endpoint"
+https://auth-$REGION.beyondidentity.com/v1/tenants/$TENANT_ID/realms/$REALM_ID/applications/$APPLICATION_ID/authorize
+```
+
+Copy the `Authorization Endpoint` from to console and append the following parameters:
+
+```bash title="/authorize"
+$(Authorization Endpoint)?
+response_type=code
+&client_id=$CLIENT_ID
+&redirect_uri=$REDIRECT_URI
+&scope=openid
+&state=$STATE
+&code_challenge_method=256
+&code_challenge=$PKCE_CODE_CHALLENGE
+```
+
+| Parameter | Value |
+| --- | --- |
+| **response_type** | code <br /><br />This parameter specifies that the authorization server should return an authorization code to the client. The code will be used by the client to request an access token.<div class= "note-message note-message-table"><p>**Grant Type** must be set to **Authorization Code** in the console application. </p></div> |
+| **client_id** | Copy and paste the **Client ID** value from your application's **External Protocol** tab. |
+| **redirect_uri** | This parameter specifies the URL to which the authorization server can send an authorization code. If a state parameter was specified in the /authorize request, then that will be echoed back to this URL as well.<br /><br /><div class= "note-message note-message-table"><p>This must match one of the **Redirect URIs** registered in your console application.</p></div> |
+| **scope** | openid |
+| **state** | This parameter is a random value generated by the client. It is included in the authorization request and returned in the response. This parameter helps the client maintain state between the request and the callback to prevent cross-site request forgery (CSRF) attacks. |
+| **code_challenge_method** | 256 <br /><PKCES256/><div class= "note-message note-message-table"><p>This value is optional. **PKCE** must be set to **S256** in the console application. If it is not set then you should not send the **code_challenge_method** or **code_challenge** parameters</p></div>|
+| **code_challenge** | This used in conjunction with **code_challenge_method** ensures that the authorization code obtained by the client can only be used by the client that originally requested it.<br /><br />You will need to store the hash of the **code_challenge** so that it can be passed to the token exchange endpoint later as a **code_verifier**.<br /><br /><div class= "note-message note-message-table"><p>This value is optional. **PKCE** must be set in the console application. If it is not set then you should not send the **code_challenge_method** or **code_challenge** parameters.</p></div> |
```

Comment:
nit: I think we should say "enabled" here instead of "set" if either `plain` or `s256` is allowed and `disabled` is not allowed.

```suggestion
| **code_challenge** | This used in conjunction with **code_challenge_method** ensures that the authorization code obtained by the client can only be used by the client that originally requested it.<br /><br />You will need to store the hash of the **code_challenge** so that it can be passed to the token exchange endpoint later as a **code_verifier**.<br /><br /><div class= "note-message note-message-table"><p>This value is optional. **PKCE** must be enabled in the console application. If it is not enabled then you should not send the **code_challenge_method** or **code_challenge** parameters.</p></div> |
```